src/types/base.js

```diff
@@ -0,0 +1,10 @@
+import {Element} from 'chart.js';
+
+export default class BaseAnnotation extends Element {
+
+  getCenterPoint(useFinalPosition) {
+    const {centerX, centerY} = this.getProps(['centerX', 'centerY'], useFinalPosition);
+    return {x: centerX, y: centerY};
+  }
+
+}
```

Comment:
I'm really not sure we should add a "layer" of inheritance for just this (and CC complaining about 2 lines repeated). As you might know it makes (undefined) property lookups a little slower, because all levels need to be checked.

Anyway, lets get 1.4.0 out first and see if that raises any issues.

Reply:
> I'm really not sure we should add a "layer" of inheritance for just this (and CC complaining about 2 lines repeated). As you might know it makes (undefined) property lookups a little slower, because all levels need to be checked.

I did because at the end of the day, all annotations will have the same code to provide the center, when normalization will end. I wasn't aware about the lookup (coming form java this is a normal case of inheritance).

> Anyway, lets get 1.4.0 out first and see if that raises any issues.

Ok, anyway this new layer should be released with version 2, therefore we have time to think about.

Reply:
removed base annotatioin